### PR TITLE
Registrating and updating admin

### PR DESCRIPTION
**Pull request by sviatoslav-krupa**
Add first and last name to devise parameter sanitizer for create and update actions.

### CODE REVIEW

File: app/views/devise/registrations/edit.html.slim [hunk 1]
```diff
@@ -0,0 +1,69 @@
+h2
+  | Edit
```
Comment (f1nwe) [inline app/views/devise/registrations/edit.html.slim]: You can write it in one row
h2 Edit.
BTW, does it stylized with your bootstrap template?, Seems that it doesn't.

File: app/views/devise/registrations/edit.html.slim [hunk 2]
```diff
@@ -0,0 +1,69 @@
+h2
+  | Edit 
+  = resource_name.to_s.humanize
+
+= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f|
```
Comment (f1nwe) [inline app/views/devise/registrations/edit.html.slim]: for forms use simple_form gem.

File: .rubocop.yml [hunk 3]
```diff
@@ -37,9 +37,11 @@ Metrics/ClassLength:
...
-
 Style/StringLiterals:
   Enabled: false
 
 Style/ClassAndModuleChildren:
-  EnforcedStyle: nested
\ No newline at end of file
+  EnforcedStyle: nested
+
+Style/ClassAndModuleChildren:
```
Comment (romaslmd) [inline .rubocop.yml]: mb one cfg at a time?

File: app/views/account/pages/show.html.slim [hunk 4]
```diff
@@ -0,0 +1,21 @@
+.container
+  h1= @page.title
+  .row
+    h3 Page type:
+    p= @page.page_type
```
Comment (romaslmd) [inline app/views/account/pages/show.html.slim]: invalid usage of bootstrap grids

File: spec/controllers/page_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe Account::PagesController, type: :controller do
+  context 'Page' do
+    let!(:page) { FactoryBot.create(:page) }
+
+    it 'Get#index' do
```
Comment (romaslmd) [inline spec/controllers/page_controller_spec.rb]: stick to the common conventions

File: spec/factories/pages.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+#  status       :integer
+#  screenshot   :string
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
+#  company_id   :integer
+#
+
+FactoryBot.define do
+  factory :page do
+    page_type { %i[bing angle.co linkedin crunchbase].sample }
```
Comment (romaslmd) [inline spec/factories/pages.rb]: why only 1 has a TLD?

Comment (igorkasyanchuk) [reply spec/factories/pages.rb]: it's OK, but move to constants in Page model

BING = 'bing'
...
Page::TYPES = [BING, ANGLECO, LINKEDIN....]

then
Page::TYPES.sample

File: app/models/page.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+#  source_url   :string
+#  status       :integer
+#  screenshot   :string
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
+#  company_id   :integer
+#
+
+class Page < ApplicationRecord
+  STATUSES = %i[active pending finished].freeze
```
Comment (igorkasyanchuk) [inline app/models/page.rb]: every value MUST be as constant too.
And you need to use this constants in all places in code

File: app/models/page.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  PAGE_TYPES = %i[bing angle.co linkedin crunchbase].freeze
+  enum status: STATUSES
```
Comment (igorkasyanchuk) [inline app/models/page.rb]: remove ENUM. Just use string column value. It's easier to debug in future and work with such.
In general enums are good only for columns like "male/female".
For statuses, page_type - use STRING column

File: app/models/user.rb [hunk 9]
```diff
@@ -1,3 +1,26 @@
...
+#  current_sign_in_ip     :inet
+#  last_sign_in_ip        :inet
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#  tenant_id              :integer
+#
+
 class User < ApplicationRecord
   ROLES = { sale: 0, admin: 1, moderator: 2 }.freeze
   enum role: ROLES
```
Comment (igorkasyanchuk) [inline app/models/user.rb]: change to String "role". I don't like magic numbers "1" for admin, etc.

File: app/views/account/pages/show.html.slim (hunk 4 above)
Comment (igorkasyanchuk) [reply app/views/account/pages/show.html.slim]: yes. 
col-md-12 needs to be added
and h1 also needs to be wrapped.

THIS IS GLOBAL PROBLEM FOR ALL OF YOU

Just try to discuss bootstrap together and understand how it works. We don't want to explain it to everyone.

File: spec/factories/pages.rb [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    title { Faker::Beer.name }
+    content_html { Faker::Lorem.paragraphs(10).join }
+    content { Faker::Lorem.paragraphs(10).join }
+    source_url { Faker::Internet.url }
+    status { %i[active pending finished].sample }
```
Comment (igorkasyanchuk) [inline spec/factories/pages.rb]: USE CONSTANTS HERE

File: spec/controllers/page_controller_spec.rb (hunk 5 above)
Comment (igorkasyanchuk) [reply spec/controllers/page_controller_spec.rb]: something like "GET #INDEX" ... just check other specs

File: db/seeds.rb [hunk 11]
```diff
@@ -1 +1,3 @@
-
+# This file should contain all the record creation needed to seed the database with its default values.
+# The data can then be loaded with the rails db:seed command (or created alongside the database with db:setup).
+FactoryBot.create_list(:page, 50)
```
Comment (igorkasyanchuk) [inline db/seeds.rb]: reduce number to 20
and remove comment above

File: app/models/user.rb [hunk 12]
```diff
@@ -22,7 +22,7 @@
 #
 
 class User < ApplicationRecord
-  ROLES = { sale: 0, admin: 1, moderator: 2 }.freeze
+  ROLES = { sale: 'sale', admin: 'admin', moderator: 'moderator' }.freeze
```
Comment (igorkasyanchuk) [inline app/models/user.rb]: REMOVE enum
and make
SALE_ROLE = 'sale'
ROLES  = [SALE_ROLE, MODERATOR_ROLE....]

File: db/migrate/20171221145626_change_role_type_in_users.rb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+class ChangeRoleTypeInUsers < ActiveRecord::Migration[5.1]
+  def change
+    change_column :users, :role, :string, default: :admin
```
Comment (igorkasyanchuk) [inline db/migrate/20171221145626_change_role_type_in_users.rb]: default admin? seriously? set to "sale"

File: lib/templates/erb/scaffold/_form.html.erb [hunk 14]
```diff
@@ -0,0 +1,13 @@
+<%%= simple_form_for(@<%= singular_table_name %>) do |f| %>
```
Comment (igorkasyanchuk) [inline lib/templates/erb/scaffold/_form.html.erb]: remove this file

**Pull request closed without merging**